util/meson-purge-includes.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+# This script deletes any -I command line arguments that are not
+# - Absolute paths.
+# - Ephemeral build directories.
+#
+# This function is necessary because Meson does not give adequate
+# control over what directories are passed in as -I search directories
+# to the C compiler. While Meson does provide |implicit_include_directories|,
+# support for this option is poor: empirically, Meson ignores this option for
+# some targerts. Doing it as a post-processing step ensures that Meson does
+# not allow improper #includes to compile successfully.
+#
+# This is run by meson as a postconf script. The following env variables will be
+# set:
+# - MESON_SOURCE_ROOT
+# - MESON_BUILD_ROOT
+
+echo "Purging superfluous -I arguments from $MESON_BUILD_ROOT."
+perl -pi -e 's#-I[^/][^@ ]+ # #g' -- "$MESON_BUILD_ROOT/build.ninja"
```

Comment:
Super nit-picking that you can probably ignore: as long as you're moving this code around, consider tweaking the regex so that it matches even if the trailing space doesn't exist, because the `-I` option was the last argument.

Reply:
I'll do this update.

One thing this did seem to miss when I examined the `build.ninja` was a `-I.` in the rule for `hw/top_earlgrey/sw/autogen/top_earlgrey.c`. I'll ensure the fix covers that case too.

Reply:
Update: No i'm not fixing the space issue - i've spent an hour on it and it keeps getting worse.

We'll fix this properly when we get a real build system.